### PR TITLE
Core/Script: fix bug that prevent double spending reset cooldown

### PR DESCRIPTION
**Pull request by GigaDev90**
With this fix, all spells cast 30 second before duel, it will not be restored

### CODE REVIEW

File: src/server/scripts/World/duel_reset.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+ * Copyright (C) 2008-2015 TrinityCore <http://www.trinitycore.org/>
```
Comment (Vincent-Michael) [inline src/server/scripts/World/duel_reset.cpp]: wtf?

File: src/server/scripts/World/duel_reset.cpp [hunk 2]
```diff
@@ -14,17 +14,19 @@
...
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
-
+#include <chrono>
 #include "ScriptMgr.h"
 #include "Player.h"
 #include "Pet.h"
 #include "SpellInfo.h"
 
+
```
Comment (Vincent-Michael) [inline src/server/scripts/World/duel_reset.cpp]: why?

File: src/server/scripts/World/duel_reset.cpp [hunk 3]
```diff
@@ -14,17 +14,19 @@
...
 class DuelResetScript : public PlayerScript
 {
     public:
         DuelResetScript() : PlayerScript("DuelResetScript") { }
-
+
```
Comment (Vincent-Michael) [inline src/server/scripts/World/duel_reset.cpp]: whitespace

File: src/server/scripts/World/duel_reset.cpp [hunk 4]
```diff
@@ -14,17 +14,19 @@
  * You should have received a copy of the GNU General Public License along
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
-
+#include <chrono>
```
Comment (Vincent-Michael) [inline src/server/scripts/World/duel_reset.cpp]: pls on end of header

**Comment by Runico** [comment]
that was annoying, Good Job!

File: src/server/scripts/World/duel_reset.cpp [hunk 5]
```diff
@@ -14,17 +14,18 @@
  * You should have received a copy of the GNU General Public License along
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
-
```
Comment (ShinDarth) [inline src/server/scripts/World/duel_reset.cpp]: revert this

File: src/server/scripts/World/duel_reset.cpp [hunk 6]
```diff
@@ -98,15 +99,34 @@ class DuelResetScript : public PlayerScript
...
         {
-            // remove cooldowns on spells that have < 10 min CD and has no onHold
-            player->GetSpellHistory()->ResetCooldowns([](SpellHistory::CooldownStorageType::iterator itr) -> bool
+            if (onStartDuel) 
             {
-                SpellInfo const* spellInfo = sSpellMgr->EnsureSpellInfo(itr->first);
-                return spellInfo->RecoveryTime < 10 * MINUTE * IN_MILLISECONDS && spellInfo->CategoryRecoveryTime < 10 * MINUTE * IN_MILLISECONDS && !itr->second.OnHold;
-            }, true);
-
+            // remove cooldowns on spells that have < 10 min CD > 30 sec and has no onHold
```
Comment (ShinDarth) [inline src/server/scripts/World/duel_reset.cpp]: indentation

File: src/server/scripts/World/duel_reset.cpp [hunk 7]
```diff
@@ -98,15 +99,34 @@ class DuelResetScript : public PlayerScript
...
+                    return spellInfo->RecoveryTime < 10 * MINUTE * IN_MILLISECONDS 
+                           && spellInfo->CategoryRecoveryTime < 10 * MINUTE * IN_MILLISECONDS 
+                           && !itr->second.OnHold
+                           && cooldownDuration > 0
+                           && ( spellInfo->RecoveryTime - cooldownDuration ) > (MINUTE / 2) * IN_MILLISECONDS;
+                }, true);
+            }
+            else
+            {
+              // remove cooldowns on spells that have < 10 min CD and has no onHold
```
Comment (ShinDarth) [inline src/server/scripts/World/duel_reset.cpp]: intendation

File: src/server/scripts/World/duel_reset.cpp [hunk 8]
```diff
@@ -19,12 +19,15 @@
...
 #include "Pet.h"
 #include "SpellInfo.h"
+#include <chrono>
 
 class DuelResetScript : public PlayerScript
 {
     public:
         DuelResetScript() : PlayerScript("DuelResetScript") { }
 
+        typedef std::chrono::system_clock Clock;
```
Comment (Shauren) [inline src/server/scripts/World/duel_reset.cpp]: you should just use SpellHistory::Clock instead of typedefing it here

**Pull request merged**